src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -346,7 +349,7 @@ public IDirectoryContents GetDirectoryContents(string subpath)
         /// characters or if <paramref name="filter" /> is an absolute path or outside the root directory specified in the
         /// constructor <seealso cref="PhysicalFileProvider(string)" />.
         /// </returns>
-        public IChangeToken Watch(string filter)
+        public IChangeToken Watch(string? filter)
```

Comment:
Similar comment about this being null - if I've seen this explicitly accepting null I'd think this will watch everything rather than nothing

Reply:
Made `filter` non-nullable

Reply:
FYI: `null` may be passed here: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs#L33

Reply:
I'd prefer we didn't call it if it's not going to do anything :-) Do not fix it in this PR though